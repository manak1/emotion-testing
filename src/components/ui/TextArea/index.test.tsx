import { userEvent } from "@storybook/testing-library"
import { composeStories } from "@storybook/testing-react"

import { render, screen } from "~/test/test-utils"

import * as stories from "./index.stories"
const { Default, Optional } = composeStories(stories)

describe("TextAreaのテスト", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Default />)
    expect(asFragment()).toMatchSnapshot()
  })
  it("入力するとonChangeが呼ばれる", () => {
    const onChange = jest.fn()
    render(<Default onChange={onChange} />)
    userEvent.type(screen.getByRole("textbox"), "テスト")
    expect(onChange).toBeCalledTimes(3)
  })
  it("任意の文字が表示されている", () => {
    render(<Optional />)
    expect(screen.getByText("任意")).toBeInTheDocument()
  })
})
