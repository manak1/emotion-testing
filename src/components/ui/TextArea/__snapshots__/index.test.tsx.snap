// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaのテスト snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-j7qwjs"
  >
    <div
      class="css-69i1ev"
    >
      <label
        class="css-1lu35ut"
        for="id"
      >
        ラベル
      </label>
    </div>
    <textarea
      class="css-1qtiooa"
      id="id"
    />
  </div>
</DocumentFragment>
`;
